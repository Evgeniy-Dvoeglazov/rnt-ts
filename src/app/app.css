.app {
  box-sizing: border-box;
  max-width: 1920px;
  min-width: 320px;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}
