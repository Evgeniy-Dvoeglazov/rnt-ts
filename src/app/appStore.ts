import { configureStore } from "@reduxjs/toolkit";
import { selectedMovieReducer } from "../store/selectedMovie/selectedMovieStore";
import { movieReducer } from "../store/movie/movieStore";
import { authReducer } from "../store/auth/authStore";

export const store = configureStore({
  reducer: {
    selectedMovie: selectedMovieReducer,
    movie: movieReducer,
    auth: authReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
