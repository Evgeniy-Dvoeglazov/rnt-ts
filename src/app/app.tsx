import "./app.css";
import SearchMoviePage from "../pages/searchMoviePage/searchMoviePage";
import AuthorizationPage from "../pages/authorizationPage/authorizationPage";
import RegistrationPage from "../pages/registrationPage/registrationPage";
import { useSelector } from "react-redux";
import { authSelector } from "../store/auth/authStore";
import { Navigate, Route, Routes } from "react-router-dom";
import NotFoundPage from "../pages/notFoundPage/notFoundPage";
import { selectedMovieSelector } from "../store/selectedMovie/selectedMovieStore";
import MovieInfoPage from "../pages/movieInfoPage/movieInfoPage";

export enum Pages {
  SearchMovie = "/",
  Authorization = "/signin",
  Registration = "/signup",
  MovieInfo = "/movie-info",
  NotFound = "*",
}

export function App() {
  const loggedIn = useSelector(authSelector);
  const selectedMovie = useSelector(selectedMovieSelector);

  return (
    <div className="app">
      <Routes>
        <Route
          path={Pages.SearchMovie}
          element={
            loggedIn ? (
              <SearchMoviePage />
            ) : (
              <Navigate to={Pages.Authorization} />
            )
          }
        />
        <Route
          path={Pages.MovieInfo}
          element={
            loggedIn && selectedMovie ? (
              <MovieInfoPage selectedMovie={selectedMovie} />
            ) : (
              <Navigate to={Pages.Authorization} />
            )
          }
        />
        <Route path={Pages.Authorization} element={<AuthorizationPage />} />
        <Route path={Pages.Registration} element={<RegistrationPage />} />
        <Route path={Pages.NotFound} element={<NotFoundPage />} />
      </Routes>
    </div>
  );
}
