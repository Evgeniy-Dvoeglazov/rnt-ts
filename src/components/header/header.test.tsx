import "@testing-library/jest-dom/jest-globals";
import { render, screen } from "@testing-library/react";
import { test, describe } from "@jest/globals";
import React from "react";
import { Provider } from "react-redux";
import Header from "./header";
import { store } from "../../app/appStore";

describe("header", () => {
  test("snapshot", () => {
    render(
      <Provider store={store}>
        <Header />
      </Provider>,
    );
    const header = screen.getByTestId(/header/i);
    expect(header).toMatchSnapshot();
  });
});
