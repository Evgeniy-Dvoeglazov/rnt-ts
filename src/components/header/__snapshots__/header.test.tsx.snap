// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header snapshot 1`] = `
<header
  class="header"
  data-testid="header"
>
  <div
    class="header__navigation"
  >
    <img
      alt="logo"
      class="logo"
    />
    <button
      class="button button_withoutBackground"
      title="Log Out"
      type="button"
      variant="withoutBackground"
    >
      Log Out
    </button>
  </div>
  <section
    class="searchMovie"
    data-testid="searchMovie"
  >
    <h1
      class="searchMovie__title"
    >
      Find your movie
    </h1>
    <form>
      <input
        class="searchMovie__input"
        data-testid="searchMovieInput"
        placeholder="Enter title"
        type="text"
      />
      <div
        class="searchMovie__buttons"
      >
        <div
          class="searchMovie__filter"
        >
          <p
            class="searchMovie__filter-description"
          >
            search by
          </p>
          <label
            class="radioButton"
          >
            <input
              checked=""
              class="radioButton__input"
              data-testid="searchModeTitle"
              mode="title"
              type="radio"
              variant="withBorder"
            />
            <span
              class="radioButton__span radioButton__span_withBorder"
            >
              title
            </span>
          </label>
          <label
            class="radioButton"
          >
            <input
              class="radioButton__input"
              data-testid="searchModeGenre"
              mode="genre"
              type="radio"
              variant="withBorder"
            />
            <span
              class="radioButton__span radioButton__span_withBorder"
            >
              genre
            </span>
          </label>
        </div>
        <button
          class="button button_withBackground"
          data-testid="searchMovieButton"
          title="Search"
          type="submit"
          variant="withBackground"
        >
          Search
        </button>
      </div>
    </form>
  </section>
</header>
`;
